test/inductor/test_aot_inductor.py

```diff
@@ -2676,6 +2676,62 @@ def forward(self, x):
         example_inputs = (torch.randn(16, 16, 16, device=self.device),)
         self.check_model(Model(), example_inputs)
 
+    def test_nested_tensor_from_jagged(self):
+        class Model(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.mlp = nn.Sequential(
+                    nn.Linear(128, 64), nn.ReLU(), nn.Linear(64, 32), nn.Sigmoid()
+                )
+
+            def forward(self, values, offsets):
+                nt = torch.nested.nested_tensor_from_jagged(values, offsets)
+                res = self.mlp(nt)
+                return res.values()
+
+        model = Model().to(device=self.device)
+
+        example_inputs_1 = (
+            torch.randn((15, 128), device=self.device),
+            torch.tensor([0, 3, 4, 10, 15], device=self.device),
+        )
+
+        # same "NT batch size", different actual amount of data
+        example_inputs_2 = (
+            torch.randn((31, 128), device=self.device),
+            torch.tensor([0, 1, 20, 25, 31], device=self.device),
+        )
+
+        # different "NT batch size"
+        example_inputs_3 = (
+            torch.randn((37, 128), device=self.device),
+            torch.tensor([0, 5, 16, 25, 29, 37], device=self.device),
+        )
+
+        with torch.no_grad(), config.patch(
+            {
+                "abi_compatible": self.abi_compatible,
+                "aot_inductor.debug_compile": True,
+            }
+        ):
+            dim0_values = Dim("dim0_values", min=1, max=128)
+            dim0_offsets = Dim("dim0_offsets", min=1, max=9)
+            dynamic_shapes = {"values": {0: dim0_values}, "offsets": {0: dim0_offsets}}
+            so_path: str = AOTIRunnerUtil.compile(
+                model, example_inputs_1, dynamic_shapes=dynamic_shapes
+            )
+        aot_inductor_module = AOTIRunnerUtil.load(self.device, so_path)
+
+        self.assertEqual(
+            aot_inductor_module(*example_inputs_1), model(*example_inputs_1)
+        )
+        self.assertEqual(
+            aot_inductor_module(*example_inputs_2), model(*example_inputs_2)
+        )
+        self.assertEqual(
+            aot_inductor_module(*example_inputs_3), model(*example_inputs_3)
+        )
```

Comment:
You can also use `check_model_with_multiple_inputs` ([example](https://github.com/pytorch/pytorch/blob/38278104537351c7b4683df8607057888360e759/test/inductor/test_aot_inductor.py#L667-L669))